client/simple/src/js/router.ts

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import { load } from "./loader.ts";
+import { Endpoints, endpoint, listen, ready, settings } from "./toolkit.ts";
+
+ready(() => {
+  document.documentElement.classList.remove("no-js");
+  document.documentElement.classList.add("js");
+
+  listen("click", ".close", function (this: HTMLElement) {
+    (this.parentNode as HTMLElement)?.classList.add("invisible");
+  });
+
+  listen("click", ".searxng_init_map", async function (this: HTMLElement, event: Event) {
+    event.preventDefault();
+    this.classList.remove("searxng_init_map");
+
+    load(() => import("./plugin/MapView.ts").then(({ default: Plugin }) => new Plugin(this)), {
+      on: "endpoint",
+      where: [Endpoints.results]
+    });
+  });
+
+  if (settings.plugins?.includes("infiniteScroll")) {
+    load(() => import("./plugin/InfiniteScroll.ts").then(({ default: Plugin }) => new Plugin()), {
+      on: "endpoint",
+      where: [Endpoints.results]
+    });
+  }
+
+  if (settings.plugins?.includes("calculator")) {
+    load(() => import("./plugin/Calculator.ts").then(({ default: Plugin }) => new Plugin()), {
+      on: "endpoint",
+      where: [Endpoints.results]
+    });
+  }
```

Comment:
That's an okay workaround for now, but in the future we might want to find a way to not re-declaring the plugin here and only declare it in the Python plugin file. I don't have a real idea how to do that at the moment though.